aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java

```diff
@@ -59,8 +59,7 @@ public FullJitterBackoffStrategy(final int baseDelay,
         public long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
                                          AmazonClientException exception,
                                          int retriesAttempted) {
-            int ceil = (retriesAttempted > MAX_RETRIES) ? maxBackoffTime :
-                    Math.min(baseDelay * (1 << retriesAttempted), maxBackoffTime);
+            int ceil =  (int)calculateExponentialDelay(retriesAttempted, baseDelay, maxBackoffTime);
```

Comment:
I'm not a huge fan of the casting between long and int - if we go over int's bound we're going to run into the same overflow problem when we do this cast.